src/pipx/main.py

```diff
@@ -279,14 +279,19 @@ def run_pipx_command(args: argparse.Namespace) -> ExitCode:  # noqa: C901
             include_injected=args.include_injected,
             skip=skip_list,
             force=args.force,
+            pip_args=pip_args,
         )
     elif args.command == "list":
+        if args.skip_maintenance:
+            # TODO: should I raise a DeprecationWarning here as well ?
+            logger.warning(
+                "Deprecation notice: `--skip-maintenance` is now a no-op, because maintenance tasks are now always skipped in `pipx list`. You can safely remove this flag from your calls."
+            )
```

Comment:
Sorry about the back and forth around this. IMO you can remove the warning, as the flag will be globally introduced via https://github.com/pypa/pipx/issues/1278.

Reply:
removed